docs/specs/shapes.md

```diff
@@ -7,6 +7,51 @@ The graphical elements are divided in 4 categories:
 * [Styles](#shape-style), that define the visual appearance of shapes
 * [Modifiers](#modifier) alter the curves of the shapes
 
+## Shape Rendering Model
+
+### Grouping and Ordering Rules
+
+* **shapes** are rendered in reverse order (bottom->top)
+* **groups** offer a scoping mechanism for transforms, styles, modifiers, and shapes
+* **transforms** change the coordinate system for all elements within their group, and transitively
+  for all other group-nested elements
+* **styles** and **modifiers** apply to all preceding shapes within the current scope,
+  including group-nested shapes
+* when **multiple styles** apply to the same shape, the shape is rendered repeatedly for each style,
+  in reverse order
+* when **multiple modifiers** apply to the same shape, they are composed in reverse order
+  (e.g. $Trim(Trim(shape))$)
+* when **multiple transforms** apply to the same shape (due to scope nesting), they compos in group
+  nesting order
+
+
+More formally:
+
+* for each $(shape, style)$ tuple, where $Index(shape) < Index(style)$ and $shape \in Scope(style)$:
+  * for each $modifier$, in increasing index order, where $Index(shape) < Index(modifier)$ and
+    $shape \in Scope(modifier)$:
+    * $shape = modifier(shape)$
+  * compute the total shape transformation by composing all transforms within the shape scope chain:
+    $$T_{shape} = \prod_{n=0}^{Scope(shape)} Transform(scope_n)$$
```

Comment:
In my PR I mention the transform alters the group's coordinate system so this point is conflicting a with that wording (as the transform doesn't need to be explicitly collected when rendering if they must be taken into account at a group level). We can rephrase after both PRs are merged to ensure consistency.

Reply:
Ack.